### PR TITLE
adds photon energy scaling due to cosmological redshift

### PR DESCRIPTION
**Pull request by wematthias**
added photon energy scaling due to cosmological redshift

### CODE REVIEW

File: cosmogrb/instruments/gbm/gbm_grb.py [hunk 1]
```diff
@@ -74,6 +74,7 @@ def _setup_source(self):
                 0.0,
                 self._duration,
                 source_function,
+                z = self.z,
```
Comment (grburgess) [inline cosmogrb/instruments/gbm/gbm_grb.py:77]: we always want to use '_' to keep properties protected from outside modification.

File: cosmogrb/instruments/gbm/gbm_grb.py [hunk 2]
```diff
@@ -145,6 +146,7 @@ def __init__(
         self._alpha = alpha
         self._ep = ep
         self._trise = trise
+        self.z = z
```
Comment (grburgess) [inline cosmogrb/instruments/gbm/gbm_grb.py:149]: see above for self._z

File: cosmogrb/instruments/gbm/gbm_grb.py [hunk 3]
```diff
@@ -174,6 +176,7 @@ def __init__(
     ):
 
         source_params = dict(peak_flux=peak_flux, alpha=alpha, ep=ep,)
+        self.z = z
```
Comment (grburgess) [inline cosmogrb/instruments/gbm/gbm_grb.py:179]: here again

File: cosmogrb/sampler/source.py [hunk 4]
```diff
@@ -317,6 +317,8 @@ def __init__(self, tstart, tstop, source_function, use_plaw_sample=False):
             25,
         )
 
+        self.z = z
```
Comment (grburgess) [inline cosmogrb/sampler/source.py:320]: modify

File: cosmogrb/sampler/source.py [hunk 5]
```diff
@@ -392,6 +394,13 @@ def _get_energy_integrated_max(self):
 
         return np.max(fluxes)
 
+    def _propagate_photons(self, photons):
+        """
+        scale photon energy due to cosmological redshift
+        """
+
+        return photons/(1+self.z)
```
Comment (grburgess) [inline cosmogrb/sampler/source.py:402]: nice! but keep the redshift private

File: cosmogrb/sampler/source.py [hunk 6]
```diff
@@ -437,6 +446,8 @@ def sample_photons(self, times):
                 self._source_function.emax,
                 self._source_function.response.effective_area_max,
             )
+        
+        return self._propagate_photons(source_frame_photons)
```
Comment (grburgess) [inline cosmogrb/sampler/source.py:450]: nice work. Have you tested the functionality to make sure it works?

File: cosmogrb/instruments/gbm/gbm_grb.py (hunk 1 above)
Comment (wematthias) [reply cosmogrb/instruments/gbm/gbm_grb.py]: Yes, I missed that here

File: cosmogrb/sampler/source.py (hunk 6 above)
Comment (wematthias) [reply cosmogrb/sampler/source.py]: looking at the plots of 
lightcurve.display_count_spectrum_source and
lightcurve.display_source
it seems to work.
unittests also run fine of course

**Pull request closed without merging**